source/material_model/simple_nonlinear.cc

```diff
@@ -0,0 +1,371 @@
+              Assert(dealii::numbers::is_finite(out.viscosities[i]),ExcMessage ("Error: Averaged viscosity is not finite."));
+
+              if (derivatives != NULL)
+                {
+                  derivatives->viscosity_derivative_wrt_strain_rate[i] = Utilities::derivative_of_weighted_p_norm_average(out.viscosities[i],volume_fractions, composition_viscosities, composition_viscosities_derivatives, viscosity_averaging_p);
+
+                  for (int x = 0; x < dim; x++)
+                    for (int y = 0; y < dim; y++)
+                      Assert (dealii::numbers::is_finite(derivatives->viscosity_derivative_wrt_strain_rate[i][x][y]),
+                              ExcMessage ("Averaged viscosity to strain-rate devrivative is not finite."));
+
+                  derivatives->viscosity_derivative_wrt_pressure[i]    = 0;
+                  Assert (dealii::numbers::is_finite(derivatives->viscosity_derivative_wrt_pressure[i]),
+                          ExcMessage ("Averaged viscosity to pressure devrivative is not finite."));
+
+                }
+            }
+          out.densities[i] = density;
+          out.thermal_expansion_coefficients[i] = thermal_expansivity;
+          // Specific heat at the given positions.
+          out.specific_heat[i] = specific_heat;
+          // Thermal conductivity at the given positions.
+          out.thermal_conductivities[i] = thermal_conductivities;
+          // Compressibility at the given positions.
+          // The compressibility is given as
+          // $\frac 1\rho \frac{\partial\rho}{\partial p}$.
+          out.compressibilities[i] = 0.0;
+          // Pressure derivative of entropy at the given positions.
+          out.entropy_derivative_pressure[i] = 0.0;
+          // Temperature derivative of entropy at the given positions.
+          out.entropy_derivative_temperature[i] = 0.0;
+          // Change in composition due to chemical reactions at the
+          // given positions. The term reaction_terms[i][c] is the
+          // change in compositional field c at point i.
+          for (unsigned int c=0; c < in.composition[i].size(); ++c)
+            out.reaction_terms[i][c] = 0.0;
+        }
+    }
+    template <int dim>
+    double
+    SimpleNonlinear<dim>::
+    reference_viscosity () const
+    {
+      return ref_visc;
+    }
+
+    template <int dim>
+    double
+    SimpleNonlinear<dim>::
+    reference_density () const
+    {
+      return densities[0];
+    }
+
+    template <int dim>
+    bool
+    SimpleNonlinear<dim>::
+    is_compressible () const
+    {
+      return false;
+    }
+
+    template <int dim>
+    void
+    SimpleNonlinear<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Compositional fields");
+      {
+        prm.declare_entry ("Number of fields", "0",
+                           Patterns::Integer (0),
```

Comment:
Yes, this should already be declared in Simulator. I am surprised this did not generate exceptions on your machine